test/multiverse/suites/rake/instrumentation_test.rb

```diff
@@ -0,0 +1,245 @@
+# This file is distributed under New Relic's license terms.
+# See https://github.com/newrelic/newrelic-ruby-agent/blob/main/LICENSE for complete details.
+# frozen_string_literal: true
+
+require_relative '../../../../lib/new_relic/agent/instrumentation/rake/instrumentation'
+
+class RakeInstrumentationTest < Minitest::Test
+  class TesterClass
+    include NewRelic::Agent::Instrumentation::Rake::Tracer
+
+    def name; 'Snake'; end
+    def timeout; 140.85; end
+  end
+
+  class ErrorClass < StandardError; end
+
+  def test_invoke_with_newrelic_tracing_happy_path
+    instance = TesterClass.new
+    instance_mock = MiniTest::Mock.new
+    canned_config = {'rake.connect_timeout': instance.timeout}
+    NewRelic::Agent::Instrumentation::Rake.stub :should_trace?, true, [instance.name] do
+      NewRelic::Agent.stub :config, canned_config do
+        NewRelic::Agent.stub :instance, instance_mock do
+          instance_mock.expect :wait_on_connect, nil, [instance.timeout]
+          NewRelic::Agent::Instrumentation::Rake.stub :before_invoke_transaction, nil do
+            NewRelic::Agent::Tracer.stub :in_transaction, nil do
+              NewRelic::Agent::Instrumentation::Rake.stub :record_attributes, nil do
+                instance.invoke_with_newrelic_tracing {}
+                instance_mock.verify
+              end
+            end
+          end
+        end
+      end
+    end
+  end
+
+  def test_invoke_with_newrelic_tracing_when_tracing_is_disabled
+    instance = TesterClass.new
+    NewRelic::Agent::Instrumentation::Rake.stub :should_trace?, false, [instance.name] do
+      # make absolutely sure the .config call is not being made
+      NewRelic::Agent.stub :config, -> { raise 'kaboom' } do
+        result = :result
+        assert_equal result, instance.invoke_with_newrelic_tracing { result }
+      end
+    end
+  end
+
+  def test_invoke_with_tracing_with_exception
+    instance = TesterClass.new
+    instance_mock = MiniTest::Mock.new
+    canned_config = {'rake.connect_timeout': instance.timeout}
+    NewRelic::Agent::Instrumentation::Rake.stub :should_trace?, true, [instance.name] do
+      error = RuntimeError.new('expected')
+      # produce the exception we want to have the method rescue
+      NewRelic::Agent.stub :config, -> { raise error } do
+        logger = MiniTest::Mock.new
+        NewRelic::Agent.stub :logger, logger do
+          logger.expect :error, nil, [/^Exception/, error]
+          instance.invoke_with_newrelic_tracing {}
+          logger.verify
+        end
+      end
+    end
+  end
+
+  def test_we_should_install_if_newrelic_rake_is_absent
+    NewRelic::LanguageSupport.stub :bundled_gem?, false, 'newrelic-rake' do
```

Comment:
This isn't for this PR, but since newrelic-rake hasn't been updated since 2014, we may want to re-evaluate how our agent reacts to this gem.